### ISSUE TITLE
Lab 8 Level 6 back() cannot pass 1 test case 

### ISSUE THREAD

**Issue by Brianliew77** [nus-cs2030/2122-s2#282]
![image](https://user-images.githubusercontent.com/99940198/162568402-8aa9515d-e0d8-4117-8579-fac0a212fac9.png)

Hi, anyone faced the same issue? Idk which part of my code does the error arise. Thanks alot!

**Comment by Lu-Yi-Fan** [comment]
When you go back(), the room you returned to did not contain any troll. Depending on how you implemented, one way you can debug is to have a public getter to get a list of what the room contains. You can run r3.tick(takeSword).back().getListOfThings() to see what r2 contains, if it is empty you need to debug from there.

**Comment by joannelxl** [comment]
hi, adding to Yi Fan's comment, maybe your back() method is not working properly, since you only went back to library and not to foyer (after the second back() method call).

**Comment by junbin-tan** [comment]
Hi @Brianliew77, might be easier to help if  you attached your code. But it might be possible that it's calling the wrong room if you implemented by using a List.

**Comment by jlee1007** [comment]
yep it seems that your back is not going to the correct room hence the issue

**Comment by chloeoxe** [comment]
Yup, I agree with the others. You should probably check your back() method, because it seems like your back() method isn't returning to the previous room, hence your output shows that you are still at the library (which has no troll) and not back at the foyer (where the troll is present).

**Comment by e0775876** [comment]
For me I just had to change my constructor to return the new room. :D 
GL!!

**Comment by mehakAS** [comment]
I had a similar issue but attempted level 6 using an ImList of Rooms. Then I had to create a couple more constructors to deal with this ImList and I had to make sure to return this.ImList<Rooms> in all the methods that return a new room. Hope that helps!!
